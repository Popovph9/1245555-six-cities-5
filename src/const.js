export const FLAT_TYPES = [`apartment`, `room`, `house`, `hotel`];

export const FLAT_OFFERS = [`Wifi`, `Heating`, `Kitchen`, `Cable TV`];

export const FLAT_PHOTOS = [`img/apartment-01.jpg`, `img/room.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`];

export const CITIES_NAMES = [`Paris`, `Cologne`, `Brussels`, `Amsterdam`, `Hamburg`, `Dusseldorf`];

export const RATING_GRADES = [`5`, `4`, `3`, `2`, `1`];

export const RATING_TITLES = [`perfect`, `good`, `not bad`, `badly`, `terribly`];

export const PIN_URL = `/img/pin.svg`;

export const PIN_ACTIVE_URL = `/img/pin-active.svg`;

export const PIN_SIZES = [30, 30];

export const RENDERED_REVIEWS = 10;

export const RENDERED_PINS = 10;

export const MAX_RENDERED_PHOTOS = 6;

export const STAR_WIDTH = 20;

export const MIN_REVIEW_LENGTH = `50`;

export const MAX_REVIEW_LENGTH = `300`;

export const CITIES_CLASS = `cities__places-list tabs__content`;

export const PLACES_CLASS = `near-places__list`;

export const CITIES_SUBCLASS = `cities`;

export const PLACES_SUBCLASS = `near-places`;

export const EMPTY_PAGE_CLASSNAME = `page__main page__main--index page__main--index-empty`;

export const MAIN_PAGE_CLASSNAME = `page__main page__main--index`;

export const ACTIVE_CLASSNAME = `locations__item-link tabs__item tabs__item--active`;

export const CASUAL_CLASSNAME = `locations__item-link tabs__item`;

export const STAR_CLASSNAME = `property__avatar-wrapper user__avatar-wrapper property__avatar-wrapper--pro`;

export const COMMON_CLASSNAME = `property__avatar-wrapper user__avatar-wrapper`;

export const OPEN_CLASSNAME = `places__options--opened`;

export const DEFAULT_SORTING = `Popular`;

export const HIGHT_PRICE_SORTING = `Price: high to low`;

export const LOW_PRICE_SORTING = `Price: low to high`;

export const RATING_SORTING = `Top rated first`;

export const DEFAULT_CITY = CITIES_NAMES[0];

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const AppRoute = {
  SIGN_IN: `/login`,
  FAVORITES: `/favorites`,
  ROOM: `/offer`,
  MAIN: `/`,
};

export const APIRoute = {
  HOTELS: `/hotels`,
  FAVORITES: `/favorite`,
  COMMENTS: `/comments`,
  NEAR: `/nearby`,
  LOGIN: `/login`,
};
