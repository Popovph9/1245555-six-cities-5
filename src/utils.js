import moment from "moment";

export const getRating = (rating, width) => {
  return Math.round(rating * width);
};

export const getReviewFormatDate = (date) => {
  return moment(date).format(`MMMM YYYY`);
};

export const getReviewDate = (date) => {
  return moment(date).format(`YYYY-MM-DD`);
};

export const extend = (a, b) => {
  return Object.assign({}, a, b);
};
