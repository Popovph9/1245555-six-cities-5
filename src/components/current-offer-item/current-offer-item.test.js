import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import CurrentOfferItem from "./current-offer-item";
import {noop} from "../../utils";
import {MOCK_REVIEWS, MOCK_CURRENT_OFFER, MOCK_FAVORITE_CURRENT_OFFER} from "../../test-data";
import {AuthorizationStatus} from "../../const";

describe(`Should CurrentOfferItem render correctly`, () => {
  const mockStore = configureStore([]);
  let CurrentOfferItemComponent = null;
  let store = null;

  it(`Should CurrentOfferItem render correctly without premium offers and casual host when the user is unauthorized`, () => {
    CurrentOfferItemComponent = renderer
    .create(
        <CurrentOfferItem
          reviews={MOCK_REVIEWS}
          currentOffer={MOCK_CURRENT_OFFER}
          changeFavorites={noop}
          redirectToRoute={noop}
          authorizationStatus={AuthorizationStatus.NO_AUTH}
          refreshOfferList={noop}
        />)
    .toJSON();

    expect(CurrentOfferItemComponent).toMatchSnapshot();
  });

  it(`Should CurrentOfferItem render correctly with premium offers and super host when the user is authorized`, () => {
    store = mockStore({
      STATE: {
        currentOffer: MOCK_FAVORITE_CURRENT_OFFER,
      },
    });

    CurrentOfferItemComponent = renderer
    .create(
        <Provider store={store}>
          <CurrentOfferItem
            reviews={MOCK_REVIEWS}
            currentOffer={MOCK_FAVORITE_CURRENT_OFFER}
            changeFavorites={noop}
            redirectToRoute={noop}
            authorizationStatus={AuthorizationStatus.AUTH}
            refreshOfferList={noop}
          />
        </Provider>)
    .toJSON();

    expect(CurrentOfferItemComponent).toMatchSnapshot();
  });

  it(`Should CurrentOfferItem render correctly without reviews`, () => {
    CurrentOfferItemComponent = renderer
    .create(
        <CurrentOfferItem
          reviews={[]}
          currentOffer={MOCK_CURRENT_OFFER}
          changeFavorites={noop}
          redirectToRoute={noop}
          authorizationStatus={AuthorizationStatus.NO_AUTH}
          refreshOfferList={noop}
        />)
    .toJSON();

    expect(CurrentOfferItemComponent).toMatchSnapshot();
  });
});
