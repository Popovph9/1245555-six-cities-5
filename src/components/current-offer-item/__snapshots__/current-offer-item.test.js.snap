// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should CurrentOfferItem render correctly Should CurrentOfferItem render correctly with premium offers and super host when the user is authorized 1`] = `
<div
  className="property__wrapper"
>
  <div
    className="property__mark"
  >
    <span>
      Premium
    </span>
  </div>
  <div
    className="property__name-wrapper"
  >
    <h1
      className="property__name"
    >
      The house among olive
    </h1>
    <button
      className="property__bookmark-button property__bookmark-button--active button"
      onClick={[Function]}
      type="button"
    >
      <svg
        className="property__bookmark-icon"
        height="33"
        width="31"
      >
        <use
          xlinkHref="#icon-bookmark"
        />
      </svg>
      <span
        className="visually-hidden"
      >
        To bookmarks
      </span>
    </button>
  </div>
  <div
    className="property__rating rating"
  >
    <div
      className="property__stars rating__stars"
    >
      <span
        style={
          Object {
            "width": "76%",
          }
        }
      />
      <span
        className="visually-hidden"
      >
        Rating
      </span>
    </div>
    <span
      className="property__rating-value rating__value"
    >
      3.8
    </span>
  </div>
  <ul
    className="property__features"
  >
    <li
      className="property__feature property__feature--entire"
    >
      house
    </li>
    <li
      className="property__feature property__feature--bedrooms"
    >
      2
       Bedrooms
    </li>
    <li
      className="property__feature property__feature--adults"
    >
      Max 
      4
       adult
      s
    </li>
  </ul>
  <div
    className="property__price"
  >
    <b
      className="property__price-value"
    >
      €
      200
    </b>
    <span
      className="property__price-text"
    >
       night
    </span>
  </div>
  <div
    className="property__inside"
  >
    <h2
      className="property__inside-title"
    >
      What's inside
    </h2>
    <ul
      className="property__inside-list"
    >
      <li
        className="property__inside-item"
      >
        Laptop friendly workspace
      </li>
      <li
        className="property__inside-item"
      >
        Washer
      </li>
      <li
        className="property__inside-item"
      >
        Towels
      </li>
    </ul>
  </div>
  <div
    className="property__host"
  >
    <h2
      className="property__host-title"
    >
      Meet the host
    </h2>
    <div
      className="property__host-user user"
    >
      <div
        className="property__avatar-wrapper user__avatar-wrapper property__avatar-wrapper--pro"
      >
        <img
          alt="Host avatar"
          className="property__avatar user__avatar"
          height="74"
          src="/img/avatar-angelina.jpg"
          width="74"
        />
      </div>
      <span
        className="property__user-name"
      >
        Angelina
      </span>
    </div>
    <div
      className="property__description"
    >
      <p
        className="property__text"
      >
        Discover daily local life in city center,
    friendly neighborhood, clandestine casino, karaoke,
    old-style artisans, art gallery and artist studio downstairs.
      </p>
    </div>
  </div>
  <section
    className="property__reviews reviews"
  >
    <h2
      className="reviews__title"
    >
      Reviews · 
      <span
        className="reviews__amount"
      >
        2
      </span>
    </h2>
    <ul
      className="reviews__list"
    >
      <li
        className="reviews__item"
      >
        <div
          className="reviews__user user"
        >
          <div
            className="reviews__avatar-wrapper user__avatar-wrapper"
          >
            <img
              alt="Reviews avatar"
              className="reviews__avatar user__avatar"
              height="54"
              src="img/avatar-angelina.jpg"
              width="54"
            />
          </div>
          <span
            className="reviews__user-name"
          >
            Angelina
          </span>
        </div>
        <div
          className="reviews__info"
        >
          <div
            className="reviews__rating rating"
          >
            <div
              className="reviews__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "100%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
          </div>
          <p
            className="reviews__text"
          >
            The house is very good, very happy, hygienic and simple living conditions around it are also very good
          </p>
          <time
            className="reviews__time"
            dateTime="2020-11-01"
          >
            November 2020
          </time>
        </div>
      </li>
      <li
        className="reviews__item"
      >
        <div
          className="reviews__user user"
        >
          <div
            className="reviews__avatar-wrapper user__avatar-wrapper"
          >
            <img
              alt="Reviews avatar"
              className="reviews__avatar user__avatar"
              height="54"
              src="img/avatar-angelina.jpg"
              width="54"
            />
          </div>
          <span
            className="reviews__user-name"
          >
            Angelina
          </span>
        </div>
        <div
          className="reviews__info"
        >
          <div
            className="reviews__rating rating"
          >
            <div
              className="reviews__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "60%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
          </div>
          <p
            className="reviews__text"
          >
            The deluxe room was a quite comfortable one with all the adequate facilities
          </p>
          <time
            className="reviews__time"
            dateTime="2020-11-01"
          >
            November 2020
          </time>
        </div>
      </li>
    </ul>
    <form
      action="#"
      className="reviews__form form"
      method="post"
    >
      <label
        className="reviews__label form__label"
        htmlFor="review"
      >
        Your review
      </label>
      <div
        className="reviews__rating-form form__rating"
        required={true}
      >
        <input
          checked={false}
          className="form__rating-input visually-hidden"
          id="5-stars"
          name="rating"
          onChange={[Function]}
          type="radio"
          value="5"
        />
        <label
          className="reviews__rating-label form__rating-label"
          htmlFor="5-stars"
          title="perfect"
        >
          <svg
            className="form__star-image"
            height="33"
            width="37"
          >
            <use
              xlinkHref="#icon-star"
            />
          </svg>
        </label>
        <input
          checked={false}
          className="form__rating-input visually-hidden"
          id="4-stars"
          name="rating"
          onChange={[Function]}
          type="radio"
          value="4"
        />
        <label
          className="reviews__rating-label form__rating-label"
          htmlFor="4-stars"
          title="good"
        >
          <svg
            className="form__star-image"
            height="33"
            width="37"
          >
            <use
              xlinkHref="#icon-star"
            />
          </svg>
        </label>
        <input
          checked={false}
          className="form__rating-input visually-hidden"
          id="3-stars"
          name="rating"
          onChange={[Function]}
          type="radio"
          value="3"
        />
        <label
          className="reviews__rating-label form__rating-label"
          htmlFor="3-stars"
          title="not bad"
        >
          <svg
            className="form__star-image"
            height="33"
            width="37"
          >
            <use
              xlinkHref="#icon-star"
            />
          </svg>
        </label>
        <input
          checked={false}
          className="form__rating-input visually-hidden"
          id="2-stars"
          name="rating"
          onChange={[Function]}
          type="radio"
          value="2"
        />
        <label
          className="reviews__rating-label form__rating-label"
          htmlFor="2-stars"
          title="badly"
        >
          <svg
            className="form__star-image"
            height="33"
            width="37"
          >
            <use
              xlinkHref="#icon-star"
            />
          </svg>
        </label>
        <input
          checked={false}
          className="form__rating-input visually-hidden"
          id="1-stars"
          name="rating"
          onChange={[Function]}
          type="radio"
          value="1"
        />
        <label
          className="reviews__rating-label form__rating-label"
          htmlFor="1-stars"
          title="terribly"
        >
          <svg
            className="form__star-image"
            height="33"
            width="37"
          >
            <use
              xlinkHref="#icon-star"
            />
          </svg>
        </label>
      </div>
      <textarea
        className="reviews__textarea form__textarea"
        id="review"
        maxLength="300"
        minLength="50"
        name="review"
        onChange={[Function]}
        placeholder="Tell how was your stay, what you like and what can be improved"
        required={true}
      />
      <div
        className="reviews__button-wrapper"
      >
        <p
          className="reviews__help"
        >
          To submit review please make sure to set 
          <span
            className="reviews__star"
          >
            rating
          </span>
           and describe your stay with at least 
          <b
            className="reviews__text-amount"
          >
            50 characters
          </b>
          .
        </p>
        <button
          className="reviews__submit form__submit button"
          disabled={true}
          onClick={[Function]}
          type="submit"
        >
          Submit
        </button>
      </div>
    </form>
  </section>
</div>
`;

exports[`Should CurrentOfferItem render correctly Should CurrentOfferItem render correctly without premium offers and casual host when the user is unauthorized 1`] = `
<div
  className="property__wrapper"
>
  
  <div
    className="property__name-wrapper"
  >
    <h1
      className="property__name"
    >
      The house among olive
    </h1>
    <button
      className="property__bookmark-button button"
      onClick={[Function]}
      type="button"
    >
      <svg
        className="property__bookmark-icon"
        height="33"
        width="31"
      >
        <use
          xlinkHref="#icon-bookmark"
        />
      </svg>
      <span
        className="visually-hidden"
      >
        To bookmarks
      </span>
    </button>
  </div>
  <div
    className="property__rating rating"
  >
    <div
      className="property__stars rating__stars"
    >
      <span
        style={
          Object {
            "width": "76%",
          }
        }
      />
      <span
        className="visually-hidden"
      >
        Rating
      </span>
    </div>
    <span
      className="property__rating-value rating__value"
    >
      3.8
    </span>
  </div>
  <ul
    className="property__features"
  >
    <li
      className="property__feature property__feature--entire"
    >
      house
    </li>
    <li
      className="property__feature property__feature--bedrooms"
    >
      2
       Bedrooms
    </li>
    <li
      className="property__feature property__feature--adults"
    >
      Max 
      4
       adult
      s
    </li>
  </ul>
  <div
    className="property__price"
  >
    <b
      className="property__price-value"
    >
      €
      200
    </b>
    <span
      className="property__price-text"
    >
       night
    </span>
  </div>
  <div
    className="property__inside"
  >
    <h2
      className="property__inside-title"
    >
      What's inside
    </h2>
    <ul
      className="property__inside-list"
    >
      <li
        className="property__inside-item"
      >
        Laptop friendly workspace
      </li>
      <li
        className="property__inside-item"
      >
        Washer
      </li>
      <li
        className="property__inside-item"
      >
        Towels
      </li>
    </ul>
  </div>
  <div
    className="property__host"
  >
    <h2
      className="property__host-title"
    >
      Meet the host
    </h2>
    <div
      className="property__host-user user"
    >
      <div
        className="property__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Host avatar"
          className="property__avatar user__avatar"
          height="74"
          src="/img/avatar-angelina.jpg"
          width="74"
        />
      </div>
      <span
        className="property__user-name"
      >
        Angelina
      </span>
    </div>
    <div
      className="property__description"
    >
      <p
        className="property__text"
      >
        Discover daily local life in city center,
    friendly neighborhood, clandestine casino, karaoke,
    old-style artisans, art gallery and artist studio downstairs.
      </p>
    </div>
  </div>
  <section
    className="property__reviews reviews"
  >
    <h2
      className="reviews__title"
    >
      Reviews · 
      <span
        className="reviews__amount"
      >
        2
      </span>
    </h2>
    <ul
      className="reviews__list"
    >
      <li
        className="reviews__item"
      >
        <div
          className="reviews__user user"
        >
          <div
            className="reviews__avatar-wrapper user__avatar-wrapper"
          >
            <img
              alt="Reviews avatar"
              className="reviews__avatar user__avatar"
              height="54"
              src="img/avatar-angelina.jpg"
              width="54"
            />
          </div>
          <span
            className="reviews__user-name"
          >
            Angelina
          </span>
        </div>
        <div
          className="reviews__info"
        >
          <div
            className="reviews__rating rating"
          >
            <div
              className="reviews__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "100%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
          </div>
          <p
            className="reviews__text"
          >
            The house is very good, very happy, hygienic and simple living conditions around it are also very good
          </p>
          <time
            className="reviews__time"
            dateTime="2020-11-01"
          >
            November 2020
          </time>
        </div>
      </li>
      <li
        className="reviews__item"
      >
        <div
          className="reviews__user user"
        >
          <div
            className="reviews__avatar-wrapper user__avatar-wrapper"
          >
            <img
              alt="Reviews avatar"
              className="reviews__avatar user__avatar"
              height="54"
              src="img/avatar-angelina.jpg"
              width="54"
            />
          </div>
          <span
            className="reviews__user-name"
          >
            Angelina
          </span>
        </div>
        <div
          className="reviews__info"
        >
          <div
            className="reviews__rating rating"
          >
            <div
              className="reviews__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "60%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
          </div>
          <p
            className="reviews__text"
          >
            The deluxe room was a quite comfortable one with all the adequate facilities
          </p>
          <time
            className="reviews__time"
            dateTime="2020-11-01"
          >
            November 2020
          </time>
        </div>
      </li>
    </ul>
  </section>
</div>
`;

exports[`Should CurrentOfferItem render correctly Should CurrentOfferItem render correctly without reviews 1`] = `
<div
  className="property__wrapper"
>
  
  <div
    className="property__name-wrapper"
  >
    <h1
      className="property__name"
    >
      The house among olive
    </h1>
    <button
      className="property__bookmark-button button"
      onClick={[Function]}
      type="button"
    >
      <svg
        className="property__bookmark-icon"
        height="33"
        width="31"
      >
        <use
          xlinkHref="#icon-bookmark"
        />
      </svg>
      <span
        className="visually-hidden"
      >
        To bookmarks
      </span>
    </button>
  </div>
  <div
    className="property__rating rating"
  >
    <div
      className="property__stars rating__stars"
    >
      <span
        style={
          Object {
            "width": "76%",
          }
        }
      />
      <span
        className="visually-hidden"
      >
        Rating
      </span>
    </div>
    <span
      className="property__rating-value rating__value"
    >
      3.8
    </span>
  </div>
  <ul
    className="property__features"
  >
    <li
      className="property__feature property__feature--entire"
    >
      house
    </li>
    <li
      className="property__feature property__feature--bedrooms"
    >
      2
       Bedrooms
    </li>
    <li
      className="property__feature property__feature--adults"
    >
      Max 
      4
       adult
      s
    </li>
  </ul>
  <div
    className="property__price"
  >
    <b
      className="property__price-value"
    >
      €
      200
    </b>
    <span
      className="property__price-text"
    >
       night
    </span>
  </div>
  <div
    className="property__inside"
  >
    <h2
      className="property__inside-title"
    >
      What's inside
    </h2>
    <ul
      className="property__inside-list"
    >
      <li
        className="property__inside-item"
      >
        Laptop friendly workspace
      </li>
      <li
        className="property__inside-item"
      >
        Washer
      </li>
      <li
        className="property__inside-item"
      >
        Towels
      </li>
    </ul>
  </div>
  <div
    className="property__host"
  >
    <h2
      className="property__host-title"
    >
      Meet the host
    </h2>
    <div
      className="property__host-user user"
    >
      <div
        className="property__avatar-wrapper user__avatar-wrapper"
      >
        <img
          alt="Host avatar"
          className="property__avatar user__avatar"
          height="74"
          src="/img/avatar-angelina.jpg"
          width="74"
        />
      </div>
      <span
        className="property__user-name"
      >
        Angelina
      </span>
    </div>
    <div
      className="property__description"
    >
      <p
        className="property__text"
      >
        Discover daily local life in city center,
    friendly neighborhood, clandestine casino, karaoke,
    old-style artisans, art gallery and artist studio downstairs.
      </p>
    </div>
  </div>
  <section
    className="property__reviews reviews"
  >
    <h2
      className="reviews__title"
    >
      Reviews · 
      <span
        className="reviews__amount"
      >
        
      </span>
    </h2>
  </section>
</div>
`;
